package/MDAnalysis/analysis/base.py

```diff
@@ -430,25 +517,219 @@ def run(self, start=None, stop=None, step=None, frames=None,
+        return ParallelExecutor(backend=backend, n_workers=n_workers, client=client)
+
+
+    def run(self, 
+            start: int = None, stop: int = None, step: int = None, 
+            frames: Iterable = None,
+            verbose: bool = None, 
+            n_workers: int = None, 
+            n_parts: int = None,
+            backend: str = 'local',
+            *, 
+            client: object = None,
+            progressbar_kwargs={},
+            ):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+        .. versionadded:: 2.2.0
+            verbose : bool, optional
+            Turn on verbosity
+        
+        progressbar_kwargs : dict, optional
+            ProgressBar keywords with custom parameters regarding progress bar position, etc; 
+            see :class:`MDAnalysis.lib.log.ProgressBar` for full list.
+
+        scheduler : str, optional 
+            Enables parallel execution of :meth:`AnalysisBase.run`.
+            Available schedulers
+              - 'dask' implies execution using `dask` processes (need to install MDAnalysis[dask] for that to work)
+              - 'multiprocessing' works with standard python multiprocessing module
+              - `None` if you want to run things locally
+        n_workers : int, optional
+            number of workers. 
+            Will become number of processes in multiprocessing.Pool(...) with `multiprocessing` scheduler,
+            and `n_workers` with `dask` scheduler.
+
+        .. versionchanged:: 2.2.0
+            Added ability to analyze arbitrary frames by passing a list of
+            frame indices in the `frames` keyword argument.
+
+        .. versionchanged:: 2.5.0
+            Add `progressbar_kwargs` parameter, 
+            allowing to modify description, position etc of tqdm progressbars
+        """
+        if progressbar_kwargs and backend not in (None, 'local'):
+            raise NotImplementedError("Can not display progressbar with non-local backend")
+
+        if n_workers is None:
+            if client: 
+                n_workers = sum(client.nthreads().values())
+            else:
+                n_workers = 1
+
+        # validate input parameters
+        n_parts = n_workers if n_parts is None else n_parts
+        
+        # reset backend if there is a pre-configured client available
+        backend = None if client is not None else backend
```

Comment:
Is this consistent with what you do above? Above, you did not allow to specify both a `client` and a `backend`.

Reply:
True -- it's a legacy of development process, removed.